### ISSUE TITLE
[CI] Rework and optimize workflows

### ISSUE THREAD

**Issue by chalin** [open-telemetry/opentelemetry.io#5786]
- Once the following lands, we should make a pass over the GH workflows to cleanup and optimize them to use `npm ci`
- This would also be an opportunity to use the trick employed in #5780 to other workflows, such as text and markdown lint jobs

## Tasks

- [ ] #5787
- [ ] ...

**Comment by svrnm** [comment]
There are 2 other things I'd like to see with the workflows, and since the issue title is generic enough for that, here they are:

* move all scripts from .github/workflows/scripts to ./scripts
* use https://github.com/peter-evans/create-pull-request/ instead of per-script custom logic to create and maintain pull requests

wdyt?

**Comment by chalin** [comment]
Yes, I've been wanting to move the workspace scripts into `./scripts` for a while now too.

I'm not familiar with that GH action, but I'm willing to give it a try! Thanks for suggesting it.

**Comment by svrnm** [comment]
> I'm not familiar with that GH action, but I'm willing to give it a try! Thanks for suggesting it.

See the community update action where it is in use. What I like about it that it is also able to update existing PRs, which would especially be helpful for the daily registry updates, so instead of having 1 per day there would be one until merged and then a new one. 

The other advantage would be that by removing the per-script logging of committing and creating PRs the scripts are easier to use outside of the workflows. 

I would try to find some time next year to start with the registry workflow and then we can look into the other ones if it is worth the effort

**Comment by vitorvasc** [comment]
@chalin @svrnm, is this still valid?

**Comment by svrnm** [comment]
> [@chalin](https://github.com/chalin) [@svrnm](https://github.com/svrnm), is this still valid?

I suspect yes, but @chalin would know better than me
